src/main/scala/za/co/absa/commons/os/OperatingSystem.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.os
+
+
+object OperatingSystem {
+
+  // adapted from https://stackoverflow.com/a/31547504/1773349
+
+  object OperatingSystems extends Enumeration {
```

Comment:
By a common convention, an enumeration type name should be a noun in a _singular_ form.
It will also make the outer object redundant, so you can flatten the type name `OperatingSystem.OperatingSystems` to just `OperatingSystem`.